plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -150,6 +157,10 @@ public ConnectorPageSource createPageSource(
                     updateResultJsonCodec);
         }
 
+        if (getColumnMappingMode(table.getMetadataEntry()) == ColumnMappingMode.NAME) {
```

Comment:
`getColumnMappingMode(table.getMetadataEntry()) != ColumnMappingMode.NONE` might be better.